tests/test_directives.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import sys
+from copy import deepcopy
+
+from flow.directives import Directives
+from flow.directives import NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS
+
+
+@pytest.fixture()
+def setUp(test_class=Directives, **kwargs):
```

Comment:
I am trying to understand the need for this instead of just `Directives(**kwargs)` or `Directives(available_directives_list)`.

Reply:
I thought that we may also wanna add tests for manual `_DirectiveItem` and we'll need to pass different arguments.

Reply:
If we do need to pass different arguments to `_DirectivesItem` can't we do that with `_DirectivesItem(**kwargs)` directly?

Reply:
I figured out that we need to initialize with `_DirectivesItem` only one time. So I've made the changes accordingly

Reply:
I am not sure if my point is getting across adequately. I don't believe there to be a reduction in complexity between `setup(_DirectivesItems, parallel=foo, serial=bar, name='example', ...)` and `_DirectivesItems(parallel=foo, serial=bar, name='example', ...)`, the second is shorter and more explicit. This is true for any class you would pass it.

Abstractions should only be employed when they reduce complexity (or code duplication) when used. Given that every abstraction has its own intrinsic complexity that means abstractions can be more complicated to use than the plain solution.